system/metrics/sinks/prometheus/prometheusSink.hpp

```diff
@@ -0,0 +1,116 @@
+/*################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+*/
+
+
+#pragma once
+
+#include "jmetrics.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+#include <thread>
+
+//including cpp-httplib single header file REST client
+//  doesn't work with format-nonliteral as an error
+//
+#if defined(__clang__) || defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#endif
+
+#undef INVALID_SOCKET
+#include "httplib.h"
+
+#if defined(__clang__) || defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
+
+#undef INVALID_SOCKET
+#include "httplib.h"
+
+#if defined(__clang__) || defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
+
+#ifdef _USE_OPENSSL
+#include <openssl/x509v3.h>
+#endif
+
+using namespace hpccMetrics;
+using namespace httplib;
+
+#ifdef PROMETHEUSSINK_EXPORTS
+#define PROMETHEUSSINK_API DECL_EXPORT
+#else
+#define PROMETHEUSSINK_API DECL_IMPORT
+#endif
+
+class PROMETHEUSSINK_API PrometheusMetricSink : public MetricSink
+{
+public:
+    explicit PrometheusMetricSink(const char *name, const IPropertyTree *pSettingsTree);
+    ~PrometheusMetricSink() override = default;
+    static constexpr const char * PROMETHEUS_REPORTER_TYPE = "prometheus";
```

Comment:
trivial - not quite sure why this isn't with the other strings (or inlined since only used once)